changelog/index.php

```diff
@@ -60,45 +63,66 @@ protected function processPR($repoName, $pr) {
 		return [$id, $data];
 	}
 
-	protected function shouldPRBeSkipped($title) {
+	protected function shouldPRBeSkipped($title)
+	{
 		if (preg_match('!^\d+(\.\d+(\.\d+))? ?(rc|beta|alpha)? ?(\d+)?$!i', $title)) {
 			return true;
 		}
 		return false;
 	}
 
 	/**
-	 * @throws Exception
+	 * Get the list of shipped apps from server head
+	 * Then compare for existing repos to check against
+	 * 
+	 * @param string $head the server head, master, stable25, stable19...
+	 * @return string[]
 	 */
-	protected function execute(InputInterface $input, OutputInterface $output)
+	protected function getReposToIterate($head = 'master')
 	{
-		$server = 'server';
-		$orgName = 'nextcloud';
+		$client = new \GuzzleHttp\Client();
+		$ghClient = new \Github\Client();
 
 		// TODO iterate over all repos
+		$shippedApps = [];
+		$orgRepositories = [];
 		$reposToIterate = [
 			"server",
 			"3rdparty",
-			"activity",
-			"circles",
-			"example-files",
-			"files_pdfviewer",
-			"files_rightclick",
-			"firstrunwizard",
-			"logreader",
-			"nextcloud_announcements",
-			"notifications",
-			"password_policy",
-			"photos",
-			"privacy",
-			"recommendations",
-			"serverinfo",
-			"survey_client",
-			"text",
-			"updater",
-			"viewer",
 		];
 
+		try {
+			$res = $client->request('GET', "https://raw.githubusercontent.com/nextcloud/server/$head/core/shipped.json");
+			$shippedApps = json_decode($res->getBody()->getContents(), true)['shippedApps'] ?? [];
+		} catch (\Exception $e) {
+			throw new Exception('Unable to fetch the shipped apps list.');
+		}
+
+		try {
+			/** @var \Github\Api\Organization $organizationApi */
+			$organizationApi = $ghClient->api('organization');
+			$paginator = new Github\ResultPager($ghClient, 50);
+			$parameters = array(self::ORG_NAME);
+			$repos = $paginator->fetchAll($organizationApi, 'repositories', $parameters);
+			$orgRepositories = array_map(fn($repo): string => $repo['name'], $repos);
```

Comment:
what's coming out here ? will there be more than the list that you have truncated above ?

Reply:
Yes, because shipped apps can also be within the server repo, so there is no repository to get PRs from for those 
e.g files_trashbin.